client/src/components/shared/Header.js

```diff
@@ -1,176 +1,83 @@
+
+  const menuItems = firstLevelRoutes
+    .map(r => makeMenuItem(r.menu.crumbs[0], r.path, r === currentRoute, 1));
+
+  const splitPath = location.pathname.split('/');
+
+  // show child items on the menu if we are at a child item, otherwise don't
+  if (currentRoute && (currentRoute.menu.crumbs.length > 1)) {
+    const parentItemIndex = firstLevelRoutes
+      .findIndex(r => r.menu.crumbs[0] === currentRoute.menu.crumbs[0]);
+    for (let i = 1;
+      (i < currentRoute.menu.crumbs.length) && (i < (splitPath.length - 1));
+      i += 1) {
+      const crumbTitle = (currentRoute.menu.crumbs[i] === constants.menuItemEntityId)
+        ? splitPath[i + 1] : currentRoute.menu.crumbs[i];
+      const childMenuItem = makeMenuItem(crumbTitle, splitPath.slice(0, i + 2).join('/'),
+        i === currentRoute.menu.crumbs.length - 1, i + 1);
+      menuItems.splice(parentItemIndex + i, 0, childMenuItem);
+    }
   }
 
-  menuItems.push({
-    link: '/compareTechnologies',
-    itemLinkBase: '/compareTechnologies',
-    text: 'Compare Technologies',
-    children: [],
-  });
+  let helpAnchor;
 
-  if (userStore.data.allowedFeatures.includes(features.reporting)) {
-    menuItems.push({
-      link: '/reporting',
-      text: 'Reporting',
-      children: [],
-    });
+  if (currentRoute) {
+    helpAnchor = string(currentRoute.menu.crumbs
+      .join(' ')
+      .replace(constants.menuItemEntityId, 'Item')).slugify().toString();
   }
 
-  if (userStore.data.allowedFeatures.includes(features.edit)) {
-    menuItems.push({
-      link: '/edit',
-      text: 'Edit',
-      children: [],
-    });
-  }
-
-  // Add breadcrumbs
-  const locationParts = location.pathname.split('/');
-  if (locationParts.length >= 3) {
-    let parentMenuItem = menuItems.find(i => i.itemLinkBase === makePath(locationParts, 1));
-    if (parentMenuItem) {
-      for (let i = 2; i < locationParts.length; i += 1) {
-        const childMenuItem = {
-          link: makePath(locationParts, i),
-          text: locationParts[i],
-          children: [],
-        };
-        parentMenuItem.children.push(childMenuItem);
-        parentMenuItem = childMenuItem;
-      }
+  if ((userStore.data.allowedFeatures.includes(features.edit))) {
+    menuItems.push(makeMenuItem('Edit', '/edit', location.pathname === '/edit', 1));
+    if ((splitPath.length > 2) && (splitPath[1] === 'edit')) {
+      const crumbTitle = decamelize(splitPath[2], ',')
+        .split(',').map(word => `${word.slice(0, 1).toUpperCase()}${word.slice(1)}`)
+        .join(' ');
+      menuItems.push(makeMenuItem(crumbTitle, splitPath.slice(0, 3).join('/'),
+        true, 2));
+      helpAnchor = string(`edit ${crumbTitle}`).slugify().toString();
+    }
+    if ((!helpAnchor) && (splitPath.length > 1) && (splitPath[1] === 'edit')) {
+      helpAnchor = 'edit';
     }
   }
 
-  const menuElements = makeMenuItems(menuItems, location);
+  console.log(currentRoute);
```

Comment:
@sfhardman 
Did you mean to leave these 2 console logs here?

Reply:
Nope, will remove them now...

Reply:
Done